resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
@bboreham Thanks. I agree that `getaddrinfo` might negate the effect of shuffling. For the typical docker use cases (all services being in the same subnet) though I see that shuffling introduces some variation. There could also be clients that doesn't use getaddrinfo().

Here is a test run of getaddrinfo() with and without shuffling from the embedded server.. note that the source IP of the container from which I ran the test and all the available dest IPs were the same in both the cases..

without shuffling, the result client gets is always the same..
````
root@451f904bc669:/# ifconfig
eth0      Link encap:Ethernet  HWaddr 02:42:ac:13:00:08  
          inet addr:172.19.0.8  Bcast:0.0.0.0  Mask:255.255.0.0

v4 address is 172.19.0.6 
v4 address is 172.19.0.7 
v4 address is 172.19.0.3 
v4 address is 172.19.0.4 
v4 address is 172.19.0.5 
````
with shuffing, for 3 runs..

````
root@53cd90f1e19a:/# ifconfig
eth0      Link encap:Ethernet  HWaddr 02:42:ac:13:00:08  
          inet addr:172.19.0.8  Bcast:0.0.0.0  Mask:255.255.0.0

v4 address is 172.19.0.3 
v4 address is 172.19.0.6 
v4 address is 172.19.0.7 
v4 address is 172.19.0.4 
v4 address is 172.19.0.5 

v4 address is 172.19.0.6 
v4 address is 172.19.0.4 
v4 address is 172.19.0.5 
v4 address is 172.19.0.7 
v4 address is 172.19.0.3

v4 address is 172.19.0.6 
v4 address is 172.19.0.3 
v4 address is 172.19.0.5 
v4 address is 172.19.0.4 
v4 address is 172.19.0.7